orchestrator/orchestrator/src/main_loop.rs

```diff
@@ -233,11 +234,33 @@ pub async fn eth_oracle_main_loop<S: Signer + 'static, CS: CosmosSigner>(
                     Err(e) => {
                         metrics::ETHEREUM_EVENT_CHECK_FAILURES.inc();
                         error!("Failed to get events for block range, Check your Eth node and Cosmos gRPC {:?}", e);
+
                         if let gravity_utils::error::GravityError::CosmosGrpcError(
                             CosmosGrpcError::TransactionFailed { tx: _, time: _ },
                         ) = e
                         {
                             delay_for(Duration::from_secs(10)).await;
+                        } else if let gravity_utils::error::GravityError::CosmosGrpcError(
```

Comment:
resync only in case this error happen

```
CosmosGrpcError(RequestError { error: Status { code: Unknown, message: "failed to execute message; message index: 0: create event vote record: non contiguous event nonce expected 15804 observed 32307 for validator 817314D0955AEA9B0A32248FAA03BC6AB00BC35A: invalid", metadata: MetadataMap { headers: {"content-type": "application/grpc", "x-cosmos-block-height": "1440776"} } } })
```

Reply:
maybe it can for that particular error pattern instead of doing for counting every request error?

Reply:
Yeah the current code is not ideal but I am not sure if there is a better way to handle this. 

Currently we cannot tell if this error is "irrecoverable" or just due to "network congestion", that is why I am using a counter and I assume it is not recoverable if it fails 10 times.

The whole oracle deserves some refactoring because the way it handles the message creation and the send (asynchrone) is prone to error when it has to deal with a lot of messages. I suspect some concurrent issues as well.